src/main/java/org/quiltmc/loader/impl/discovery/ModResolver.java

```diff
@@ -1712,6 +1719,8 @@ public ModBreakage(Logger logger, ModLoadOption source, ModDependency publicDep,
 		ModBreakage put(DependencyHelper<LoadOption, ModLink> helper) throws ContradictionException {
 			LoadOption[] disallowed = new LoadOption[invalidOptions.size()];
 			for (int i = 0; i < invalidOptions.size(); i++) {
+				// Obtain the root of the invalid options to resolve mod breakages that occur because a mod that breaks
+				// is provided.
 				disallowed[i] = invalidOptions.get(i).getRoot();
```

Comment:
Technically the reason is that we don't want to have multiple "variables" for a single mod - as a provided mod is always present if the providing mod is present, and always missing if the providing mod is not present. Using `getRoot` ensures we actually depend on the provider (and thus we solve correctly) rather than potentially saying that a "provided" mod is present but the provider isn't (or vice versa)

Reply:
I put your description in there since it seems to be more comprehensible than mine.